lib/screens/v2/candidate_generator/widgets/reconstructed_alert.ex

```diff
@@ -157,21 +159,43 @@ defmodule Screens.V2.CandidateGenerator.Widgets.ReconstructedAlert do
          location_context: location_context,
          fetch_stop_name_fn: fetch_stop_name_fn,
          is_terminal_station: is_terminal_station,
-         now: now
+         now: now,
+         fetch_subway_platforms_for_stop_fn: fetch_subway_platforms_for_stop_fn
        ) do
     Enum.map(alerts, fn alert ->
+      all_platforms_at_informed_station =
+        get_platforms_at_informed_station(alert, fetch_subway_platforms_for_stop_fn)
+
       %ReconstructedAlert{
         screen: config,
         alert: alert,
         now: now,
         location_context: location_context,
         informed_stations: get_stations(alert, fetch_stop_name_fn),
         is_terminal_station: is_terminal_station,
-        is_full_screen: is_full_screen
+        is_full_screen: is_full_screen,
+        all_platforms_at_informed_station: all_platforms_at_informed_station
```

Comment:
Yeah this is something I struggled with while writing it and can't find think of a solution I'm super happy with. Would having a `context` like in `SubwayStatus` make this any clearer?

Reply:
Could this struct have a `is_partial_station_closure` field instead, and we can make that determination all in one place? "Exporting" the final value seems easier to follow than doing half the logic here, "exporting" an intermediate value with a highly specific meaning, then doing the other half later in another place.

Reply:
The issue is needing the `platform_name` [here](https://github.com/mbta/screens/blob/288958ab56590f92becccd910ab47f4e91deb9aa/lib/screens/v2/widget_instance/reconstructed_alert.ex#L1094). So just knowing if the alert is a partial closure isn't enough for this one case.

Reply:
Then maybe `partial_closure_platform_names`? It seems we're already encoding "presentational" data in this struct since `informed_stations` is just a list of station names, so this would be in line with that. And the name would make it clearer that this is only populated when the alert is a partial station closure. Whatever we can do to put the "final result" we need into the struct here, in one place, I think will be better than splitting up the logic.

Somewhat related thought: what happens when an alert partially-closes multiple stations? It's a definite possibility within Alerts UI.

Reply:
Had a chat with Dave about the multiple stations question. He is ok working under the assumption that the scenario won't happen. If it does, it can use the full station closure logic.